server/src/main/java/org/apache/iotdb/db/mpp/aggregation/CountAccumulator.java

```diff
@@ -36,28 +37,31 @@ public CountAccumulator() {}
 
   // Column should be like: | Time | Value |
   @Override
-  public int addInput(Column[] column, TimeRange timeRange) {
+  public int addInput(Column[] column, IWindow curWindow) {
+    int windowControlColumnIndex = curWindow.getControlColumnIndex();
+    int curPositionCount = column[windowControlColumnIndex].getPositionCount();
+
     TimeColumn timeColumn = (TimeColumn) column[0];
     Column valueColumn = column[1];
     long minTime = Math.min(timeColumn.getStartTime(), timeColumn.getEndTime());
     long maxTime = Math.max(timeColumn.getStartTime(), timeColumn.getEndTime());
-    if (!valueColumn.mayHaveNull() && timeRange.contains(minTime, maxTime)) {
+
+    if (curWindow.isTimeWindow()
+        && !valueColumn.mayHaveNull()
+        && ((TimeWindow) curWindow).getCurTimeRange().contains(minTime, maxTime)) {
```

Comment:
```suggestion
        && curWindow.contains(column[0])) {
```
add a new method in IWindow.

Reply:
same as all other Accumulator

Reply:
I've added a new method `contains()` in IWindow.